cache.js

```diff
@@ -0,0 +1,31 @@
+module.exports = function Cache(config) {
+   return {
+      getCachedResponse: function(req) {
+         return new Promise(function(resolve, reject) {
+            req.fromCache = false;
+            req.cacheKey = getCacheKey(req.session, req.body, req.parserOptions);
+            resolve(null);
+         })
+      },
+
+      setCachedResponse: function(req, css) {
+         if (req.fromCache) {
+            return css;
+         }
+         return new Promise(function(resolve, reject) {
+            resolve(css);
+         })
+      }
+   }
+}
+
+function getCacheKey(session, lessCode, parseOptions) {
+   return md5([lessCode, JSON.stringify(parseOptions) || '']);
+}
+
+function md5(strings) {
```

Comment:
This function already exists in a slightly different form:
https://github.com/iFixit/blessyou/blob/4d915da7e5e868f80571f74a1688fcf6c4ba7892/sessions.js#L62-L67

It looks like you could just map over the array in `getCacheKey`?

Reply:
I was going by the rule of three. This is only the second repetition and it's only a few lines (and it's different).

Also, the other function would end up using this one cause this creates a hash of concatenated strings.